### DISCUSSION TITLE
How to commit to two branches from an action?

### DISCUSSION THREAD

**Discussion by jakubmikita** [community/community#26187] [Actions]
<p>I’d like to create a <code>git-flow</code> action which releases a new version.</p>
<p>It basically used two branches: develop and master, creates the release branch from develop, then merge it to develop and master and tags the new version. Ref: <a href="https://danielkummer.github.io/git-flow-cheatsheet/index.html" rel="nofollow noopener">https://danielkummer.github.io/git-flow-cheatsheet/index.html</a></p>
<p>Current workflow (simplified):</p>
<pre><code>name: Release new version

on: repository_dispatch

jobs:
  release:
    name: Release
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2
        with:
          ref: develop
      - name: Fetch master branch
        run: |
          git fetch origin master
          git checkout --track origin/master
          git pull origin master --allow-unrelated-histories
      - name: Install git-flow
        run: |
          sudo apt-get install git-flow -y
      - name: Start release
        run: |
          git flow init -d
          git flow release start ${{ github.event.client_payload.new_version }}
      - name: Replace [Next] tags with new version number
        uses: jacobtomlinson/gha-find-replace@master
        with:
          find: "\\[Next\\]"
          replace: "${{ github.event.client_payload.new_version }}"
      - name: Commit version bump
        run: |
          git commit -am "Version bump"
      - name: Finish release
        run: |
          git flow release finish ${{ github.event.client_payload.new_version }}
</code></pre>
<p>But when I try to finish the release and commit to master I’m getting the</p>
<pre><code>refusing to merge unrelated histories
</code></pre>
<p>error.</p>
<p>Workflow run link: <a href="https://github.com/BracketSpace/Notification/runs/471806605?check_suite_focus=true" rel="nofollow noopener">https://github.com/BracketSpace/Notification/runs/471806605?check_suite_focus=true</a></p>
<p>How can I commit to both develop and master branches in one action?</p>
<p>Tried with two actions/checkout@v2 steps but it makes only one branch available.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/kubitomakita">@kubitomakita</a> ,</p>
<p>You can use ‘actions/checkout@v1’ instead, checkout the 2 branches one by one. I confirmed with ‘git branch’, both branch is listed.</p>
<p>Besides, please remove ‘/’ for ‘/readme.txt’ in action ‘jacobtomlinson/gha-find-replace’, otherwise the stable tag won’t be replaced.</p>
<p>Thanks.</p>

**Comment by jakubmikita** [comment]
<p>Thanks for the help! Changed the action to v1 indeed helped, but with additional clean parameter.</p>
<p>I’m going to compose the complete solution</p>

**Comment by jakubmikita** [comment, marked as answer]
<p>As <a class="mention" href="/weide-zhou">@weide-zhou</a> suggested, changing the checkout action to v1 helps.</p>
<p>Working config:</p>
<pre><code>steps:
  - name: Checkout master
    uses: actions/checkout@v1
    with:
      ref: 'master'
  - name: Checkout develop
    uses: actions/checkout@v1
    with:
      ref: 'develop'
      clean: false
</code></pre>
<p>Run: <a href="https://github.com/BracketSpace/Notification/runs/473033135?check_suite_focus=true" rel="nofollow noopener">https://github.com/BracketSpace/Notification/runs/473033135?check_suite_focus=true</a></p>
<p>I’m not sure why it doesn’t work with v2. Maybe the clean action is not working properly?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/kubitomakita">@kubitomakita</a> ,</p>
<p>This is due to ‘actions/checkout@v2’ will delete the prior branch. You can find the details from the execution log below:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3ba5972073c34893363f909c98a199a7674364fd.png" data-download-href="/uploads/short-url/8vEWG2Ji427Q8KXvgQopMUX8ozP.png?dl=1" title="checkoutv2.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181095990-5de2e4ce-1f1c-4402-b947-65892c32ae84.png" alt="checkoutv2.png" data-base62-sha1="8vEWG2Ji427Q8KXvgQopMUX8ozP" width="690" height="404" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3ba5972073c34893363f909c98a199a7674364fd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">checkoutv2.png</span><span class="informations">915×537 32.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jakubmikita** [comment]
<p>Yes exactly <a class="mention" href="/weide-zhou">@weide-zhou</a> but weird enough, the “clean: false” shouldn’t remove it.</p>

**Comment by weide-zhou** [comment]
<p>According to the ‘checkout’ usage, clean parameter is related to below operation but not related to branch deletion.</p>
<pre><code># Whether to execute `git clean -ffdx &amp;&amp; git reset --hard HEAD` before fetching# Default: trueclean:''
</code></pre>
<p>If you still have any doubt, it’s recommended to raise a ticket in the checkout issue <a href="https://github.com/actions/checkout/issues" rel="nofollow noopener">list</a> for confirmation.</p>
<p>Thanks.</p>
